sqlmesh/utils/pydantic.py

```diff
@@ -1,32 +1,125 @@
 import typing as t
+from functools import wraps
 
-from pydantic import BaseModel
+import pydantic
+from pydantic.fields import FieldInfo
 from sqlglot import exp
 
+from sqlmesh.core import dialect as d
+
+if t.TYPE_CHECKING:
+    Model = t.TypeVar("Model", bound="PydanticModel")
+
+
 DEFAULT_ARGS = {"exclude_none": True, "by_alias": True}
 
 
-class PydanticModel(BaseModel):
-    class Config:
-        arbitrary_types_allowed = True
-        extra = "forbid"
-        json_encoders = {
-            exp.Expression: lambda e: e.meta.get("sql") or e.sql(dialect=e.meta.get("dialect"))
-        }
-        underscore_attrs_are_private = True
-        smart_union = True
+PYDANTIC_MAJOR_VERSION = int(pydantic.__version__.split(".")[0])
+
+
+if PYDANTIC_MAJOR_VERSION >= 2:
+
+    def field_validator(*args: t.Any, **kwargs: t.Any) -> t.Callable[[t.Any], t.Any]:
+        kwargs.pop("always", None)
```

Comment:
Consider adding a comment why the kwarg is ignored (default behavior now or something else?)